kernel/aster-nix/src/ipc/semaphore/system_v/sem.rs

```diff
@@ -0,0 +1,307 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use core::{
+    sync::atomic::{AtomicU64, Ordering},
+    time::Duration,
+};
+
+use ostd::sync::{Mutex, Waiter, Waker};
+
+use super::sem_set::SEMVMX;
+use crate::{
+    ipc::{key_t, semaphore::system_v::sem_set::sem_sets, IpcFlags},
+    prelude::*,
+    process::{Pid, Process},
+    time::{
+        clocks::{RealTimeCoarseClock, JIFFIES_TIMER_MANAGER},
+        timer::Timeout,
+    },
+};
+
+#[derive(Clone, Copy, Debug, Pod)]
+#[repr(C)]
+pub struct SemBuf {
+    sem_num: u16,
+    sem_op: i16,
+    sem_flags: i16,
+}
+
+impl SemBuf {
+    pub fn sem_num(&self) -> u16 {
+        self.sem_num
+    }
+
+    pub fn sem_op(&self) -> i16 {
+        self.sem_op
+    }
+
+    pub fn sem_flags(&self) -> i16 {
+        self.sem_flags
+    }
+}
+
+#[repr(u16)]
+#[derive(Debug)]
+enum Status {
+    Normal = 0,
+    Pending = 1,
+    Removed = 2,
+}
+
+struct PendingOp {
+    sem_buf: SemBuf,
+    status: Arc<Mutex<Status>>,
```

Comment:
Using `Mutex` for `Status` may be heavy-weight. Can we use `AtomicStatus` instead? 
```rust
pub struct AtomicStatus(AtomicU8)
```